src/vulkan-renderer/wrappers/instance.cpp

```diff
@@ -0,0 +1,174 @@
+            instance_extension_wishlist.push_back(requested_instance_extension.c_str());
+        }
+    }
+
+    // We are not checking for duplicated entries but this is no problem.
+    for (const auto &instance_extension : instance_extension_wishlist) {
+        if (availability_checks->has_instance_extension(instance_extension)) {
+            spdlog::debug("Adding '{}' to list of enabled instance extensions.", instance_extension);
+            enabled_instance_extensions.push_back(instance_extension);
+        } else {
+            spdlog::error("Requested instance extension '{}' is not available on this system!", instance_extension);
+        }
+    }
+
+    std::vector<const char *> instance_layers_wishlist = {};
+
+#ifndef NDEBUG
+    // RenderDoc is a very useful open source graphics debugger for Vulkan and other APIs.
+    // Not using it all the time during development is fine, but as soon as something crashes
+    // you should enable it, take a snapshot and look up what's wrong.
+    if (enable_renderdoc_instance_layer) {
+        instance_layers_wishlist.push_back("VK_LAYER_RENDERDOC_Capture");
+    }
+
+    // We can't stress enough how important it is to use validation layers during development!
+    // Validation layers in Vulkan are in-depth error checks for the application's use of the API.
+    // They check for a multitude of possible errors. They can be disabled easily for releases.
+    // Understand that in contrary to other APIs, in Vulkan API the driver provides no error checks
+    // for you! If you use Vulkan API incorrectly, your application will likely just crash.
+    // To avoid this, you must use validation layers during development!
+    if (enable_validation_layers) {
+        spdlog::debug("Vulkan validation layers are enabled.");
+        spdlog::debug("Adding 'VK_LAYER_KHRONOS_validation' to instance extension wishlist.");
+        instance_layers_wishlist.push_back("VK_LAYER_KHRONOS_validation");
+    } else {
+        spdlog::warn("Vulkan validation layers are not enabled although debug configuration is selected!");
+        spdlog::warn("You must always use validation layers during development if you are serious about writing stable software.");
+        spdlog::warn("Without them, most bugs are literally impossible to find.");
+        spdlog::warn("Even worse, your software might work on your machine but does crash on other systems!");
+    }
+#endif
+
+    // Add requested instance layers to wishlist.
+    if (requested_instance_layers.has_value()) {
+        for (const auto &instance_layer : requested_instance_layers.value()) {
+            instance_layers_wishlist.push_back(instance_layer.c_str());
+        }
+    }
+
+    // We have to check which instance layers of our wishlist are available on the current system!
+    // We are not checking for duplicated entries but this is no problem.
+    for (auto current_layer : instance_layers_wishlist) {
```

Comment:
We should make this `const auto *` to be more explicit that's it's a pointer (and not copying).

Reply:
ok.

Reply:
I will be using
```
for (const auto& current_layer : instance_layers_wishlist)
```

Reply:
done.